infrastructures/infrastructure-common/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/ConnectorIaasJSONTransformer.java

```diff
@@ -56,6 +56,32 @@ public static String getInfrastructureJSONWithEndPoint(String infrastructureId,
         return infrastructure.toString();
     }
 
+    public static String getOpenstackInfrastructureJSONWithEndPoint(String infrastructureId, String type,
+            String username, String password, String domain, String scopePrefix, String scopeValue, String region,
+            String identityVersion, String endpoint, boolean toBeRemovedOnShutdown) {
+        JSONObject credentials = new JSONObject();
+        credentials.put("username", username);
+        credentials.put("password", password);
+        credentials.put("domain", domain);
+        JSONObject scope = new JSONObject();
+        scope.put("prefix", scopePrefix);
+        scope.put("value", scopeValue);
+
+        JSONObject infrastructure = new JSONObject().put("id", infrastructureId)
+                                                    .put("type", type)
+                                                    .put("credentials", credentials)
+                                                    .put("scope", scope)
+                                                    .put("region", region)
+                                                    .put("identityVersion", identityVersion)
+                                                    .put("toBeRemovedOnShutdown", toBeRemovedOnShutdown);
+
+        if (endpoint != null && !endpoint.isEmpty()) {
```

Comment:
should it be the same check for all other passed String parameters?

Reply:
Checking removed, thanks